core/src/apps/wallet/sign_tx/dash.py

```diff
@@ -0,0 +1,470 @@
+        ofs = 0
+        proregtx_payload = bytes()
+        while ofs < proregtx.extra_data_len:
+            size = min(1024, proregtx.extra_data_len - ofs)
+            chunk = await helpers.request_tx_extra_data(tx_req, ofs, size, initial_proregtx_id)
+            proregtx_payload += chunk
+            ofs += len(chunk)
+        ownerkeyid_position = _varint_size(proregtx_payload) + 60
+        owner_keyid = proregtx_payload[ownerkeyid_position:ownerkeyid_position + 20]
+        if keyid_from_sig != owner_keyid:
+            raise ProcessError("Payload signature doesn't match Owner key")
+
+    async def _parse_pro_up_rev_tx(self, data, position):
+        version = unpack("<H", data[position:position + 2])[0]
+        if not version == 1:
+            raise ProcessError("Unknown Dash Provider Update Registrar format version")
+        position += 2
+        initial_proregtx_id = bytes(reversed(data[position:position + 32]))
+        position += 32
+        tx_req = TxRequest()
+        tx_req.details = TxRequestDetailsType()
+        proregtx = await helpers.request_tx_meta(tx_req, initial_proregtx_id)
+        if proregtx.version != ((1 << 16) | 3):
+            raise ProcessError("Invalid ProRegTx")
+        self.confirmations.extend([("Initial ProRegTx", _to_hex(initial_proregtx_id))])
+        reason = unpack("<H", data[position:position + 2])[0]
+        position += 2
+        self.confirmations.extend([("Revoke reason", _revoke_reason(reason))])
+        if data[position:position + 32] != self.inputs_hash:
+            raise ProcessError("Invalid inputs hash in DIP2 transaction")
+        position += 32
+        if position + _BLS_SIGNATURE_SIZE != len(data):
+            raise ProcessError("Invalid payload BLS signature size")
+
+    def _parse_cb_tx(self, data, position):
+        raise ProcessError("Unsupported Dash DIP3 transaction type")
+
+    def _parse_qm_tx(self, data, position):
+        raise ProcessError("Unsupported Dash DIP3 transaction type")
+
+    def _parse_bu_reg_tx(self, data, position):
+        raise ProcessError("Unsupported Dash DIP3 transaction type")
+
+    def _parse_bu_credit_tx(self, data, position):
+        raise ProcessError("Unsupported Dash DIP3 transaction type")
+
+    def _parse_bu_reset_tx(self, data, position):
+        raise ProcessError("Unsupported Dash DIP3 transaction type")
+
+    def _parse_bu_close_tx(self, data, position):
+        raise ProcessError("Unsupported Dash DIP3 transaction type")
+
+
+async def confirm_tx_detail(ctx, title, data):
+    text = Text(title, ui.ICON_SEND, icon_color=ui.GREEN)
+    text.bold(data)
+    return await require_confirm(ctx, text, ButtonRequestType.SignTx)
+
+
+# Used to check if this transaction requires special processing
+def is_dip2_tx(tx):
+    if not tx.coin_name.lower().startswith("dash"):
+        return False
+    version = tx.version
+    dip2_type = version >> 16
+    version &= 0xffff
+    return (version is 3) and (dip2_type > 0)
```

Comment:
Maybe missing something, but shouldn't `(version is 3)` be `(version >= 3)` since future updates may increase the version number?

Reply:
We don't know anything about version 4 transactions and we don't know really if it is a correct way to process this version 4 transactions,  so we call this functionality only for version 3 transactions. It is the same condition like in the Core,  here,  for example: https://github.com/dashpay/dash/blob/master/src/evo/specialtx.cpp#L22